### PR TITLE
resource_control: reset the resource statistics when limiter is recreated 

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: ref #14900

What's Changed:

```
Add a auto-increment version for resource limiter to track the resource limiter changes and reset the statistics data when version is changed.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/resource_control/src/worker.rs [hunk 1]
```diff
@@ -196,12 +217,12 @@ impl<R: ResourceStatsProvider> GroupQuotaAdjustWorker<R> {
         // the available resource for background tasks is defined as:
         // (total_resource_quota - foreground_task_used). foreground_task_used
         // resource is calculated by: (resource_current_total_used -
-        // background_consumed_total). We reserve 10% of the free resources for
+        // background_consumed_total). We reserve 20% of the free resources for
```
Comment (Connor1996) [inline components/resource_control/src/worker.rs:220]: why

**Review by Connor1996 — APPROVED** [review]
LGTM

File: components/resource_control/src/worker.rs (hunk 1 above)
Comment (glorv) [reply components/resource_control/src/worker.rs:220]: In my test, 10% is not enough to restrict the resource usage very well, this should due to our under calculation of the resource usage, so I adjust it to 20% and the result is much better in my benchmark.

**Review by nolouch — APPROVED** [review]
lgtm

File: components/resource_control/src/worker.rs (hunk 1 above)
Comment (glorv) [reply components/resource_control/src/worker.rs]: Rolled it back because I originally want to do this change in #15166

**Pull request merged**